examples/sol-contract/src/processor.rs

```diff
@@ -0,0 +1,107 @@
+//! Program instruction processor
+//! Only the program admin can issue the Init instruction.
+//! And anyone can check the loan with the Loan2Value instruction.
+
+use solana_program::account_info::{
+    next_account_info,
+    AccountInfo,
+};
+use solana_program::entrypoint::ProgramResult;
+use solana_program::msg;
+use solana_program::program_error::ProgramError;
+use solana_program::program_pack::{
+    IsInitialized,
+    Pack,
+};
+use solana_program::pubkey::Pubkey;
+
+use borsh::BorshDeserialize;
+use pyth_sdk_solana::load_price_feed_from_account_info;
+
+use crate::instruction::ExampleInstructions;
+use crate::state::LoanInfo;
+
+pub fn process_instruction(
+    _program_id: &Pubkey,
+    _accounts: &[AccountInfo],
+    input: &[u8],
+) -> ProgramResult {
+    let account_iter = &mut _accounts.iter();
+    let signer = next_account_info(account_iter)?;
+    let data_account = next_account_info(account_iter)?;
+    let pyth_loan_account = next_account_info(account_iter)?;
+    let pyth_collateral_account = next_account_info(account_iter)?;
+
+    let instruction = ExampleInstructions::try_from_slice(input)?;
+    match instruction {
+        ExampleInstructions::Init {} => {
+            if !(signer.key == _program_id && signer.is_signer) {
+                return Err(ProgramError::Custom(0));
+            }
+
+            let mut loan_info = LoanInfo::unpack_from_slice(&data_account.try_borrow_data()?)?;
+
+            if loan_info.is_initialized() {
+                return Err(ProgramError::Custom(1));
+            }
+
+            loan_info.is_initialized = true;
+            loan_info.loan_key = *pyth_loan_account.key;
+            loan_info.collateral_key = *pyth_collateral_account.key;
+            // Give some dummy numbers for simplicity of this example.
+            loan_info.loan_qty = 1;
+            loan_info.collateral_qty = 3000;
```

Comment:
Good point. I have made these two numbers parameters of the `Loan2Value` instruction. And then I modified `invoke.ts` so that the client will pass these numbers to the on-chain program.